include/seqan3/alignment/configuration/align_config_aligned_ends.hpp

```diff
@@ -0,0 +1,600 @@
+
+#include <type_traits>
+#include <tuple>
+
+#include <seqan3/alignment/configuration/detail.hpp>
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/core/metafunction/template_inspection.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+
+namespace seqan3::align_cfg
+{
+
+/*!\brief A mixin class which can maintain a static or a dynamic bool state.
+ * \ingroup configuration
+ * \tparam value_t    The value type to be represented. Must be of type std::true_type, std::false_type or `bool`.
+ * \tparam _is_static A boolean that evaluates to true only if `value_t` is a std::integral_constant.
+ * \tparam _value     A boolean that captures the value of `value_t`, iff `_is_static` is `true`. Otherwise false.
+ *
+ * \details
+ *
+ * This mixin base class provides an optional pattern regarding the static state of the represented value.
+ * If the mixin is constructed from an std::integral_constant it will hold an static state of the wrapped value.
+ * In the other case, when constructing it from a boolean, the state of the value will be dynamic.
+ */
+template <typename value_t = bool,
+          bool _is_static  = std::conditional_t<std::is_same_v<value_t, bool>, std::false_type, std::true_type>::value,
+          bool _value      = std::conditional_t<_is_static, value_t, std::false_type>::value>
+//!\cond
+    requires std::Same<value_t, std::true_type> || std::Same<value_t, std::false_type> || std::Same<value_t, bool>
+//!\endcond
+class seq_end_gap_base
+{
+protected:
+
+    //!\brief Friend of this class to provide access to static constexpr members.
+    template <typename ...ends_t>
+    friend class end_gaps;
+
+    //!\brief Used to differentiate between static and dynamic state.
+    static constexpr bool is_static    = _is_static;
+    //!\brief Holds the static value if the state is static.
+    static constexpr bool static_value = _value;
+
+    /*!\name Constructor, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr seq_end_gap_base()                                     noexcept = default;
+    constexpr seq_end_gap_base(seq_end_gap_base const &)             noexcept = default;
+    constexpr seq_end_gap_base(seq_end_gap_base &&)                  noexcept = default;
+    constexpr seq_end_gap_base & operator=(seq_end_gap_base const &) noexcept = default;
+    constexpr seq_end_gap_base & operator=(seq_end_gap_base &&)      noexcept = default;
+    ~seq_end_gap_base()                                              noexcept = default;
+
+    /*!\brief Constructs the wrapped value from the passed object.
+     * \tparam    _value_t Either a std::integral_constant or bool.
+     * \param[in] v        The value to wrap.
+     */
+    template <typename _value_t>
+    explicit constexpr seq_end_gap_base(_value_t const v) noexcept
+    {
+        if constexpr (std::is_same_v<_value_t, bool>)
+            value = v;
+        else
+            value = v();
```

Comment:
>You can still just assign inside the constructor. 

That's true, that should work since it is implicitly convertible to bool. 

> Altogether, I think two different overloards for the class template would be more readable and probably also save a lot of code, because they could be aggregates.

Splitting the mixin is already tracked as followup issue but it would not change the semantics. So it does not need be split now. In general I am relatively certain that I can make this an aggregate as is and would bring the spliting into another sprint.